percona/tests/upstream_update/performance_test.go

```diff
@@ -0,0 +1,217 @@
+	stdDevHwmPerc  float64
+
+	meanData        float64
+	stdDevDataBytes float64
+	stdDevDataPerc  float64
+}
+
+func TestPerformance(t *testing.T) {
+	if !getBool(doRun) {
+		t.Skip("For manual runs only through make")
+		return
+	}
+
+	var updated, original *StatsData
+	t.Run("upstream exporter", func(t *testing.T) {
+		updated = doTestStats(t, repeatCount, scrapesCount, updatedExporterFileName)
+	})
+
+	t.Run("percona exporter", func(t *testing.T) {
+		original = doTestStats(t, repeatCount, scrapesCount, oldExporterFileName)
+	})
+
+	printStats(original, updated)
+}
+
+func calculatePerc(base, updated float64) float64 {
+	diff := base - updated
+	diffPerc := float64(100) / base * diff
+	diffPerc = diffPerc * -1
+
+	return diffPerc
+}
+
+func doTestStats(t *testing.T, cnt int, size int, fileName string) *StatsData {
+	var durations []float64
+	var hwms []float64
+	var datas []float64
+
+	for i := 0; i < cnt; i++ {
+		d, hwm, data, err := doTest(size, fileName)
+		if !assert.NoError(t, err) {
+			return nil
+		}
+
+		durations = append(durations, float64(d))
+		hwms = append(hwms, float64(hwm))
+		datas = append(datas, float64(data))
+	}
+
+	mean, _ := stats.Mean(durations)
+	stdDev, _ := stats.StandardDeviation(durations)
+	stdDev = float64(100) / mean * stdDev
+
+	clockTicks, err := sysconf.Sysconf(sysconf.SC_CLK_TCK)
+	if err != nil {
+		panic(err)
+	}
+
+	mean = mean * float64(1000) / float64(clockTicks) / float64(size)
+	stdDevMs := stdDev / float64(100) * mean
+
+	meanHwm, _ := stats.Mean(hwms)
+	stdDevHwm, _ := stats.StandardDeviation(hwms)
+	stdDevHwmPerc := float64(100) / meanHwm * stdDevHwm
+
+	meanData, _ := stats.Mean(datas)
+	stdDevData, _ := stats.StandardDeviation(datas)
+	stdDevDataPerc := float64(100) / meanData * stdDevData
+
+	st := StatsData{
+		meanMs:     mean,
+		stdDevMs:   stdDevMs,
+		stdDevPerc: stdDev,
+
+		meanHwm:        meanHwm,
+		stdDevHwmBytes: stdDevHwm,
+		stdDevHwmPerc:  stdDevHwmPerc,
+
+		meanData:        meanData,
+		stdDevDataBytes: stdDevData,
+		stdDevDataPerc:  stdDevDataPerc,
+	}
+
+	//fmt.Printf("loop %dx%d: sample time: %.2fms [deviation ±%.2fms, %.1f%%]\n", cnt, scrapesCount, st.meanMs, st.stdDevMs, st.stdDevPerc)
+	fmt.Printf("running %d scrapes %d times\n", size, cnt)
+	fmt.Printf("CPU\t%.1fms [±%.1fms, %.1f%%]\n", st.meanMs, st.stdDevMs, st.stdDevPerc)
+	fmt.Printf("HWM\t%.1fkB [±%.1f kB, %.1f%%]\n", st.meanHwm, st.stdDevHwmBytes, st.stdDevHwmPerc)
+	fmt.Printf("Data\t%.1fkB [±%.1f kB, %.1f%%]\n", st.meanData, st.stdDevDataBytes, st.stdDevDataPerc)
+
+	return &st
+}
+
+func doTest(iterations int, fileName string) (cpu, hwm, data int64, _ error) {
+	cmd, port, collectOutput, err := launchExporter(fileName)
+	if err != nil {
+		return 0, 0, 0, err
+	}
+
+	total1 := getCPUTime(cmd.Process.Pid)
+
+	for i := 0; i < iterations; i++ {
+		_, err = tryGetMetrics(port)
+		if err != nil {
+			return 0, 0, 0, errors.Wrapf(err, "Failed to perform test iteration %d.%s", i, collectOutput())
```

Comment:
```suggestion
			return 0, 0, 0, errors.Wrapf(err, "failed to perform test iteration %d.%s", i, collectOutput())
```

Reply:
And in all other places same: https://github.com/golang/go/wiki/CodeReviewComments#error-strings